components/browser/session/src/main/java/mozilla/components/browser/session/storage/DefaultSessionStorage.kt

```diff
@@ -140,20 +141,21 @@ class DefaultSessionStorage(
     internal fun serializeSession(session: Session): JSONObject {
         return JSONObject().apply {
             put("url", session.url)
+            put("source", session.source.name)
         }
     }
 
     @Throws(JSONException::class)
     internal fun deserializeSession(id: String, json: JSONObject): Session {
-        return Session(json.getString("url"), id)
+        return Session(json.getString("url"), Source.valueOf(json.getString("source")), id)
```

Comment:
This can get problematic if we decide to rename a source. Do you think we should handle the case where there's no matching Source? And/or should we pass a String to the enum constants so that the string representation is  independent from the name of the enum constants?

Reply:
I think going with the constants and their string representation is easier than adding a secondary String. Let's handle the case where we don't find a matching name and fall back to Source.NONE. Even if we add a secondary string for persistence, there's no guarantee that we won't forget to keep its original name :). I will update this.

Reply:
👍

Reply:
@pocmo updated and test added.